src/test/scala/org/broadinstitute/hail/methods/SkatoSuite.scala

```diff
@@ -0,0 +1,387 @@
+
+    property("handle splat") =
+      forAll(compositeAnnotationGen) {case (vds, annotations, genes, phenotypes, annotType) =>
+        val tmpOutput = tmpDir.createTempFile("splatTest")
+        val variantAnnotations = vds.variants.zip(sc.parallelize(annotations)).map{case (v, va) => (v, Annotation(va))}
+        val phenotypeMap = vds.sampleIds.zip(phenotypes.map{case p => Annotation(p.orNull)}).toMap
+
+        val groupKeyQueryString =
+          if (annotType == "Set")
+            s"let x = va.vep.transcript_consequences.map(csq => csq.gene_id).toSet in if (x.size == 0) NA: Set[String] else x"
+          else
+            s"let x = va.vep.transcript_consequences.map(csq => csq.gene_id) in if (x.length == 0) NA: Array[String] else x"
+
+        val splatResults = tmpOutput + ".splat.tsv"
+        val noSplatResults = tmpOutput + ".nosplat.tsv"
+        val configFile = tmpOutput + ".config"
+
+        var s = State(sc, sqlContext, vds
+          .annotateVariants(variantAnnotations, testVepSignature, List("vep"))
+          .annotateSamples(phenotypeMap, sampleSignature, List("pheno")))
+        s = SplitMulti.run(s, Array.empty[String])
+
+        val nNonMissingPheno = phenotypes.count(_.isDefined)
+        val nVariants = s.vds.nVariants
+
+        writeConfig(tmpOutput + ".config")
+
+        val cmdNoSplat = mutable.ArrayBuffer("--config", configFile ,
+          "-k", groupKeyQueryString,
+          "-y","sa.pheno.phenotype"
+        )
+        val cmdSplat = mutable.ArrayBuffer("--config", configFile,
+          "-k", groupKeyQueryString,
+          "-y","sa.pheno.phenotype"
+        )
+
+        cmdNoSplat += "-o"
+        cmdNoSplat += noSplatResults
+
+        cmdSplat += "-o"
+        cmdSplat += splatResults
+        cmdSplat += "--splat"
+
+        if (nNonMissingPheno == 0 || nVariants == 0) {
+          try {
+            s = GroupTestSKATO.run(s, cmdNoSplat.result().toArray)
+            s = GroupTestSKATO.run(s, cmdSplat.result().toArray)
+            false
+          } catch {
+            case e: FatalException => true
+            case _:Throwable => false
+          }
+        } else {
+          s = GroupTestSKATO.run(s, cmdNoSplat.result().toArray)
+          s = GroupTestSKATO.run(s, cmdSplat.result().toArray)
+
+          val (baseType, querier) = s.vds.queryVA(groupKeyQueryString)
+
+          val (nSplatGroupsAnswer, nNoSplatGroupsAnswer) = baseType match {
+            case _: TSet =>
+              val trueGroups = s.vds.rdd.flatMap{case (v, va, gs) => querier(va)}.map(_.asInstanceOf[Set[String]]).collect()
+              val nSplatGroupsAnswer = trueGroups.foldLeft(Set.empty[String])((comb, s) => comb.union(s)).size
+              val nNoSplatGroupsAnswer = trueGroups.toSet.size
+              (nSplatGroupsAnswer, nNoSplatGroupsAnswer)
+            case _: TArray =>
+              val trueGroups = s.vds.rdd.flatMap{case (v, va, gs) => querier(va)}.map(_.asInstanceOf[IndexedSeq[String]]).collect()
```

Comment:
You can unify these two cases by casting to `Iterable[String]`, which is a common superclass of Set and IndexedSeq.